src/test/java/itest/ShouldAcceptMultipartWithBoundaryIT.java

```diff
@@ -0,0 +1,183 @@
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (TimeoutException e) {
+            System.out.println("Test hangs out during cleanup. Reason: " + e.getMessage());
+            throw new RuntimeException("Cleanup did not complete within the expected time.", e);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof HttpException) {
+                HttpException httpException = (HttpException) e.getCause();
+                MatcherAssert.assertThat(httpException.getStatusCode(), Matchers.equalTo(400));
+                MatcherAssert.assertThat(httpException.getMessage(), Matchers.equalTo("Bad Request"));
+            } else {
+                // Log the actual cause of the execution exception
+                System.out.println("Cleanup execution failed. Reason: " + e.getCause().getMessage());
+                throw e;
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            System.out.println("Cleanup operation interrupted. Reason: " + e.getMessage());
+            throw e;
+        }
+    }
+    @Test
+    @Order(1)
+    void shouldAcceptMultipartWithBoundaryDefault() throws Exception {
+        testMultipartFormData("multipart/form-data");
+    }
+
+    @Test
+    @Order(2)
+    void shouldAcceptMultipartWithBoundarySpecific() throws Exception {
+        testMultipartFormData("multipart/form-data; boundary=------somecharacters");
+    }
+
+    @Test
+    @Order(3)
+    void shouldAcceptMultipartWithBoundaryUnknown() throws Exception {
+        testMultipartFormData("multipart/form-data; unkown characters");
+    }
+
+    @Test
+    @Order(4)
+    void shouldAcceptMultipartWithBoundaryDirectives() throws Exception {
+        testMultipartFormData("multipart/form-data; directive1; directive2");
+    }
+
+    @Test
+    @Order(5)
+    void shouldAcceptMultipartWithBoundarySingleDirective() throws Exception {
+        testMultipartFormData("multipart/form-data; directive");
+    }
+
+    void testMultipartFormData(String contentType) throws ExecutionException, InterruptedException, java.util.concurrent.TimeoutException {
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.set("name", "multipart");
+        form.set("matchExpression", "false");
+        form.set("eventSpecifier", "template=Continuous");
+    
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), contentType)
+                .sendForm(form, ar -> {
+                    assertRequestStatus(ar, response);
+                });
+    
+                try {
+                    JsonObject result = response.get(5, TimeUnit.SECONDS);
```

Comment:
The `response` future is always going to time out here I think, I don't see anywhere that it gets `.complete()`d with a value.